artiq/coredevice/comm_kernel.py

```diff
@@ -616,9 +617,10 @@ def service(obj, attr, value): return setattr(obj, attr, value)
                 self._write_int32(embedding_map.store_str(function))
             else:
                 exn_type = type(exn)
-                if exn_type in (ZeroDivisionError, ValueError, IndexError, RuntimeError) or \
-                        hasattr(exn, "artiq_builtin"):
-                    name = "0:{}".format(exn_type.__name__)
+                if hasattr(builtins, exn_type.__qualname__):
+                    name = "0:{}".format(exn_type.__qualname__)
+                elif hasattr(exn, "artiq_builtin"):
+                    name = "0:{}.{}".format(exn_type.__module__, exn_type.__qualname__)
```

Comment:
> Why was this changed?

The name here is used to fetch corresponding entry from the EmbeddingMap (https://github.com/m-labs/artiq/blob/990df7fa775c2df7cbacc2779cb7869365ad5965/artiq/coredevice/comm_kernel.py#L629), so the names must match the names in `str_forward_map` otherwise a new id is allocated. This additional check is needed because ARTIQ exceptions are named differently from the Python core exceptions.

Reply:
> So this is fixing a bug unrelated to this PR? Maybe split it into separate commit or PR then?

will split this into a separate commit then. It should still be a part of this PR as previously, it would generate a new ID for the exception which may or may not be mapped in `eh_artiq`.